### PR TITLE
CI: Test numpy 1.24 on 3.11 build

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by lithomas1** [comment]
Why don't we try unpinning bumpy on 3.11 instead? That build has no numba IIRC.

**Comment by phofl** [comment]
Hm did not consider this, thanks for pointing it out. Makes sense to me

**Comment by phofl** [comment]
cc @mroeschke this is ready for a review now, not sure if we want to leave all the ignores in...

File: ci/deps/actions-311.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ dependencies:
 
   # required dependencies
   - python-dateutil
-  - numpy<1.24.0
+  - numpy>=1.24.0
```
Comment (mroeschke) [inline ci/deps/actions-311.yaml:21]:
```suggestion
  - numpy
```

(Until we enforce the same min version on these files)

File: pandas/tests/series/test_constructors.py [hunk 2]
```diff
@@ -804,7 +804,7 @@ def test_constructor_floating_data_int_dtype(self, frame_or_series):
         # Long-standing behavior (for Series, new in 2.0 for DataFrame)
         #  has been to ignore the dtype on these;
         #  not clear if this is what we want long-term
-        expected = frame_or_series(arr)
+        # expected = frame_or_series(arr)
```
Comment (mroeschke) [inline pandas/tests/series/test_constructors.py:807]: Are these supposed to be commented out?

File: pandas/core/arrays/masked.py [hunk 3]
```diff
@@ -428,10 +428,14 @@ def to_numpy(
                     "for this dtype."
                 )
             # don't pass copy to astype -> always need a copy since we are mutating
-            data = self._data.astype(dtype)
+            with warnings.catch_warnings():
+                warnings.filterwarnings("ignore")
```
Comment (mroeschke) [inline pandas/core/arrays/masked.py:432]: It would be good to now what type of warnings these `warnings.filterwarnings("ignore")` are

**Comment by mroeschke** [comment]
> not sure if we want to leave all the ignores in

If there are any deprecation or future warnings those should be fixed

File: pandas/core/arrays/masked.py (hunk 3 above)
Comment (phofl) [reply pandas/core/arrays/masked.py:432]: All runtime warnings iirc, but can add

File: pandas/tests/series/test_constructors.py (hunk 2 above)
Comment (phofl) [reply pandas/tests/series/test_constructors.py:807]: Yes, they were unused, but the comment above is a bit confusing, that’s why I did not delete them
Outright

**Comment by phofl** [comment]
We are only filtering RuntimeWarnings now.

**Comment by mroeschke** [comment]
Were any of the RuntimeWarnings addressable, or were they specific to the data being passed?

**Comment by phofl** [comment]
All data dependent.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Comment by mroeschke** [comment]
I think these tests are failing due to numpy being unpinned here

```
 FAILED pandas/tests/plotting/test_series.py::TestSeriesPlots::test_bar_log - AssertionError: numpy array are different

numpy array values are different (16.66667 %)
[left]:  [0.09999999999999999, 1.0, 10.0, 100.0, 1000.0, 10000.0]
[right]: [0.1, 1.0, 10.0, 100.0, 1000.0, 10000.0]
FAILED pandas/tests/plotting/frame/test_frame_subplots.py::TestDataFramePlotsSubplots::test_bar_log_no_subplots - AssertionError: numpy array are different

numpy array values are different (25.0 %)
[left]:  [0.09999999999999999, 1.0, 10.0, 100.0]
[right]: [0.1, 1.0, 10.0, 100.0]
FAILED pandas/tests/plotting/frame/test_frame_subplots.py::TestDataFramePlotsSubplots::test_bar_log_subplots - AssertionError: numpy array are different

numpy array values are different (16.66667 %)
[left]:  [0.09999999999999999, 1.0, 10.0, 100.0, 1000.0, 10000.0]
[right]: [0.1, 1.0, 10.0, 100.0, 1000.0, 10000.0]
```

**Comment by phofl** [comment]
Yikes this is weird, I xfailed them but this build started failing because the tests were passing again

**Pull request merged**